homeassistant/components/otbr/util.py

```diff
@@ -258,3 +264,22 @@ async def update_issues(
     """Raise or clear repair issues related to network settings."""
     await _warn_on_channel_collision(hass, otbrdata, dataset_tlvs)
     _warn_on_default_network_settings(hass, otbrdata, dataset_tlvs)
+
+
+async def update_unique_id(
+    hass: HomeAssistant, entry: ConfigEntry | None, extended_address: bytes
+) -> None:
+    """Update the config entry's unique_id if not matching."""
+    extended_address_hex = extended_address.hex()
+    if (
+        entry
+        and entry.source == SOURCE_USER
+        and entry.unique_id != extended_address_hex
+    ):
+        _LOGGER.debug(
+            "Updating unique_id of entry %s from %s to %s",
+            entry.entry_id,
+            entry.unique_id,
+            extended_address_hex,
```

Comment:
I wonder if we should use the border agent id here instead (`id` in the DNS-SD TXT record) :thinking: 

Thread specification 1.3.0 has both fields `xa` and `id` as reserved, so relying on this to match the mDNS record is a bit "out of spec" technically :cold_sweat: 

`xa` stands for extended address, and represents the MAC address of the Thread interface of the border router. It is 64-bit long. To the OTBR code base it got introduced in 2021 with https://github.com/openthread/ot-br-posix/pull/1111. However, technically, it is not part of the Thread 1.3.0. But all Thread Border Routers seem to announce it (Google, Apple, Nanoleaf, Espressif's ESP TBR, and likely others)

`id` is the border agent ID, which is a random generated ID 128-bit long. To the OTBR code base it got introduced in 2023 with https://github.com/openthread/ot-br-posix/pull/1832. Therefor, our add-on has it since a while. As well as the Espressif ESP TBRs which emulate the same REST API and can therefor be configured by this integration. Most BRs have that field, but Apple BRs today don't announce this field via mDNS.

We do use `xa` in our WS API as primary way to identify a BR (since https://github.com/home-assistant/core/pull/108282). From what I remember the reason we went for `xa` there was since the frontend anyway has to rely on `xa` since Apple BRs don't have the `id` field just yet.

From what I understand, the `xa` changes on every network reconfiguration. Whereas the `id` is supposed to stay constant (unless a factory reset is made). Our "Reset border router" command from the Thread panel does a factory reset, so both change anyways in that case.

So effectively, it doesn't really matter all that much for us. We also need to be able to handle a change, at least when a OTBR factory reset is done. But the `id` might be a bit more stable in the future, so it might be the better choice :thinking: